pkg/cfg/sanitizer.go

```diff
@@ -0,0 +1,47 @@
+package cfg
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+)
+
+type Sanitizer func(in interface{}) (interface{}, error)
+
+func Sanitize(in map[string]interface{}, sanitizers []Sanitizer) (interface{}, error) {
+	var err error
+
+	if len(sanitizers) == 0 {
+		return in, nil
+	}
+
+	for key, val := range in {
+		for _, san := range sanitizers {
+			if in[key], err = san(val); err != nil {
+				return nil, fmt.Errorf("can not apply sanitizer on key %s: %w", key, err)
+			}
+		}
+
+		if valMap, ok := val.(map[string]interface{}); ok {
+			if in[key], err = Sanitize(valMap, sanitizers); err != nil {
+				return nil, fmt.Errorf("can not apply sanitizer on key %s: %w", key, err)
+			}
+		}
+	}
+
+	return in, nil
+}
+
+func TimeSanitizer(in interface{}) (interface{}, error) {
+	if reflect.TypeOf(in) != reflect.TypeOf(time.Time{}) {
+		return in, nil
+	}
+
+	tm, ok := in.(time.Time)
+
+	if !ok {
```

Comment:
Will this branch be reachable? You already check the value type of the input to be of `time.Time`, why should the type assertion fail?

Reply:
well, i think one could skip that..